### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by SebastianStehle**
I have implemented the test improvements with strategy pattern. I actually found a bug in sqlite. Therefore the tests are not green yet.

### CODE REVIEW

File: tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj [hunk 1]
```diff
@@ -13,8 +13,10 @@
           <PrivateAssets>all</PrivateAssets>
           <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>
+        <PackageReference Include="FluentAssertions" Version="7.2.0" />
```
Comment (PhenX) [inline tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj:16]: I'm not very used to fluent assertions nor like the fluent syntax for this usage, was it necessary ?

File: tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/TestHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        else if (entities.Count > 0)
+        {
+            runId = entities[0].TestRun;
+        }
+
+        var actualConfigure = configure ?? (_ => { });
+        switch (strategy)
+        {
+            case InsertStrategy.InsertReturn:
```
Comment (PhenX) [inline tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/TestHelpers.cs:51]: My thought was to have two enums or two bools, but it works well like this too, as long as there are no more than two parameters

File: tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj (hunk 1 above)
Comment (SebastianStehle) [reply tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj:16]: it is easier not to miss something. If you just test one property you can easily miss edge cases where your assertions are really bad.

Comment (PhenX) [reply tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj:16]: This was more a problem from the quality of my assertions, not the lack of fluent assertions.
That and the recent licensing issues with FluentAssertions, I was not ready for this.
Next time you want to integrate a new dependency I'd like to discuss about it.

**Comment by PhenX** [comment]
Good catch for the sqlite issue!

**Comment by SebastianStehle** [comment]
Yeah, when you find one bug with something like it is worth it :)

**Comment by SebastianStehle** [comment]
Tests are green again. I am done here.

File: tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj (hunk 1 above)
Comment (PhenX) [reply tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/PhenX.EntityFrameworkCore.BulkInsert.Tests.csproj:16]: No problem, by the way I love Verify!

**Review by PhenX — APPROVED** [review]
